### PR TITLE
Add user release benchmark so that we can run it on pull request

### PR DESCRIPTION
**Pull request by atalman**

### CODE REVIEW

File: .github/workflows/userbenchmark-a100-release.yml
Comment (xuzhao9) [inline .github/workflows/userbenchmark-a100-release.yml:45]: Nit: L45-L54 is not needed, since this workflow is only for release test

**Comment by xuzhao9** [comment]
LGTM, we could submit a test PR after this PR gets merged, and see if it will work.

In addition, we could add `userbenchmark/` path to `path-ignore/` at https://github.com/pytorch/benchmark/blob/main/.github/workflows/pr-test.yml#L5, this will save some CI resource by skip testing the models on userbenchmark changes.

**Review by xuzhao9 — APPROVED** [review]

File: .github/workflows/userbenchmark-a100-release.yml
Comment (xuzhao9) [inline .github/workflows/userbenchmark-a100-release.yml:42]: It seems `version.txt` is under `userbenchmark/release-test/version.txt`?

Comment (atalman) [inline .github/workflows/userbenchmark-a100-release.yml:42]:
```suggestion
          release_version=$(cat userbenchmark/release-test/version.txt)
```

Comment (atalman) [reply .github/workflows/userbenchmark-a100-release.yml]: Yes true, let me try

**Pull request closed without merging**
